mattermost/mattermost-webapp
# [MM-36280] Show recent searches in Search popover (#18841)

#### Summary
Show recent searches of the user is the search bar.

**QA**
- Make a search query
- Reopen the search bar
- Select a search type: file or message
- Make sure that the recent searches are shown in the list
- Make sure that the list is navigable with the mouse also the keyboard (up and down arrows)
- Make sure that the behavoir of the recent search items matches the description in [Figma](https://www.figma.com/file/SwCd4uBAt0gAWnGpgjfO92/MM-29044-File-Search-Enhancements?node-id=87%3A14946)

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-36280

#### Related Pull Requests

- This PR in the server side could be done seperately and once it's merged we can add the remove button (https://github.com/mattermost/mattermost-server/pull/21148)

#### Screenshots

|  before  |  after  |
|----|----|
| ![image](https://user-images.githubusercontent.com/36013798/193467677-6db26e98-90e7-45a4-88a2-01bcf8cd0bd0.png) | ![image](https://user-images.githubusercontent.com/36013798/193467664-d18d1c23-4b91-4ae5-9157-eab25bcf2af4.png) |

#### Release Note

```release-note
Show 5 most recent searches in the search popover
```
